File: app/src/main/java/com/dominikgaller/alpinebooking/booking/core/domain/exception/InvalidBookingStateException.java
package com.dominikgaller.alpinebooking.booking.core.domain.exception;

import com.dominikgaller.alpinebooking.booking.core.domain.TourBookingStatus;

/**
 * Thrown when a state transition is attempted on a booking that is not in
 * the required state.
 *
 * <p>Maps to HTTP 409 Conflict at the REST boundary.
 *
 * <p>SDD: See {@code documentation/use-cases/uc02-confirm-tour-booking.spec.md}, section 9.
 */
public class InvalidBookingStateException extends RuntimeException {

    public InvalidBookingStateException(final TourBookingStatus currentStatus) {
        super("Cannot confirm booking in state: " + currentStatus.name());
    }
}

File: src/main/kotlin/com/example/contractmanagement/booking/core/domain/tourbooking/exception/InvalidBookingStateException.kt
package com.example.contractmanagement.booking.core.domain.tourbooking.exception

import com.example.contractmanagement.booking.core.domain.tourbooking.TourBookingStatus

/**
 * Thrown when a state transition is attempted from a state that does not permit it.
 *
 * Maps to HTTP 409 (`coding-style.definition.md` § 6.2).
 */
class InvalidBookingStateException(
    currentStatus: TourBookingStatus,
) : RuntimeException("Invalid state transition from state: $currentStatus")
